.github/workflows/std-workflows.yml

```diff
@@ -18,12 +20,16 @@ jobs:
         with:
           username: ${{ secrets.DOCKER_USERNAME }}
           password: ${{ secrets.DOCKER_ACCESS_TOKEN }}
+      
+      - name: Get Hash
+        id: get_commit_hash
+        run: echo "git_hash=$(git rev-parse --short "$GITHUB_SHA")" >> $GITHUB_ENV
 
       - name: Building flask application.
         run: |
           make docker-build
         env:
-          IMAGE_TAG: ${{secrets.IMAGE_TAG}}
+          IMAGE_TAG: ${{ env.git_hash }}
```

Comment:
You defined a env variable and used it in other places. This is fine way to work with. 
But just for you knowledge, Github actions also supports outputs. https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-defining-outputs-for-a-job